lib/astarte/client/credentials.ex

```diff
@@ -0,0 +1,155 @@
+    expiry = Keyword.get(opts, :expiry, @default_expiry)
+
+    Credentials.new()
+    |> append_appengine_claim(@api_all_access_claim_value)
+    |> append_channels_claim("JOIN::.*")
+    |> append_channels_claim("WATCH::.*")
+    |> set_expiry(expiry)
+  end
+
+  def pairing_all_access_credentials(opts \\ []) do
+    expiry = Keyword.get(opts, :expiry, @default_expiry)
+
+    Credentials.new()
+    |> append_pairing_claim(@api_all_access_claim_value)
+    |> set_expiry(expiry)
+  end
+
+  def realm_management_all_access_credentials(opts \\ []) do
+    expiry = Keyword.get(opts, :expiry, @default_expiry)
+
+    Credentials.new()
+    |> append_realm_management_claim(@api_all_access_claim_value)
+    |> set_expiry(expiry)
+  end
+
+  def housekeeping_all_access_credentials(opts \\ []) do
+    expiry = Keyword.get(opts, :expiry, @default_expiry)
+
+    Credentials.new()
+    |> append_housekeeping_claim(@api_all_access_claim_value)
+    |> set_expiry(expiry)
+  end
+
+  def astartectl_credentials(opts \\ []) do
+    expiry = Keyword.get(opts, :expiry, @default_expiry)
+
+    Credentials.new()
+    |> append_appengine_claim(@api_all_access_claim_value)
+    |> append_realm_management_claim(@api_all_access_claim_value)
+    |> append_pairing_claim(@api_all_access_claim_value)
+    |> append_flow_claim(@api_all_access_claim_value)
+    |> set_expiry(expiry)
+  end
+
+  def append_housekeeping_claim(%Credentials{} = credentials, claim) when is_binary(claim) do
+    append_claim(credentials, "a_ha", claim)
+  end
+
+  def append_realm_management_claim(%Credentials{} = credentials, claim) when is_binary(claim) do
+    append_claim(credentials, "a_rma", claim)
+  end
+
+  def append_pairing_claim(%Credentials{} = credentials, claim) when is_binary(claim) do
+    append_claim(credentials, "a_pa", claim)
+  end
+
+  def append_appengine_claim(%Credentials{} = credentials, claim) when is_binary(claim) do
+    append_claim(credentials, "a_aea", claim)
+  end
+
+  def append_channels_claim(%Credentials{} = credentials, claim) when is_binary(claim) do
+    append_claim(credentials, "a_ch", claim)
+  end
+
+  def append_flow_claim(%Credentials{} = credentials, claim) when is_binary(claim) do
+    append_claim(credentials, "a_f", claim)
+  end
+
+  defp append_claim(%Credentials{claims: claims} = credentials, claim_key, claim_value) do
+    updated_claims = Map.update(claims, claim_key, [claim_value], &[claim_value | &1])
+    %{credentials | claims: updated_claims}
+  end
+
+  def set_expiry(%Credentials{} = credentials, :infinity) do
+    %{credentials | expiry: :infinity}
+  end
+
+  def set_expiry(%Credentials{} = credentials, expiry_seconds)
+      when is_integer(expiry_seconds) do
+    %{credentials | expiry: expiry_seconds}
+  end
+
+  def to_jwt(%Credentials{claims: astarte_claims, expiry: expiry}, private_key_pem) do
+    base_config =
+      %{}
+      |> Joken.Config.add_claim("iss", fn -> "Astarte Cloud" end)
```

Comment:
Should we change issuer?

Reply:
Put `Astarte Client` for now, we probably are going to make it configurable in a second iteration